lib/src/util.dart

```diff
@@ -96,6 +98,52 @@ String normalizeLinkLabel(String label) {
   return text;
 }
 
+///  Decodes HTML entity and numeric character references, for example decode
+/// `&#35` to `#`.
+String decodeHtmlCharacters(String input) =>
+    input.replaceAllMapped(htmlCharactersPattern, decodeHtmlCharacterFromMatch);
+
+/// Decodes HTML entity and numeric character references from the given [match].
+String decodeHtmlCharacterFromMatch(Match match) {
+  final text = match.match;
+  final isEntity = match[1] != null;
+  final isDecimal = match[2] != null;
+  final isHexadecimal = match[3] != null;
+  String? decodedText;
+
+  // Entity references, see
+  // https://spec.commonmark.org/0.30/#entity-references.
+  if (isEntity) {
+    decodedText = htmlEntitiesMap[text];
```

Comment:
Here we can return `htmlEntitiesMap[text] ?? text`.

Reply:
Fixed!